### PR TITLE
Feature fe prospects page 15

### PR DESCRIPTION
**Pull request by mcgraths7**
**What it does**

1. Creates overall page layouts with reusable DrawerAndContent component -> just pass in the components as props

2. Creates a reusable table for displaying and sorting data

3. Creates a reusable search component which can be used for filtering data

4. Provides a barebones prospects page -> has a search on the drawer, and the table on the content side. Search is wired up via context which wraps the page

**What to look out for**

1. The new file structure splits each page into "content", "drawer", and "page" components. The page component passes the drawer and content components into the reusable DrawerAndContent component.

**Known issues**

1. This is going to conflict with another PR #46 . I will resolve the conflicts once they inevitably come out during merging

- Changed overall file structure
- Changed form library
- Changed how routes work

### CODE REVIEW

**Review by mcgraths7 — COMMENTED** [review]
Resolved all conflicts

File: server/.gitignore [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+.vscode
+
+# Coverage reports
+htmlcov/
+.coverage
+.coverage.*
+*,cover
+
+#pipfiles
+Pipfile
```
Comment (Patton-L) [inline server/.gitignore]: I think pipfile shouldn't be ignore

File: server/Pipfile [hunk 2]
```diff
@@ -1,19 +0,0 @@
-[[source]]
```
Comment (Patton-L) [inline server/Pipfile]: Pipfile shouldn't be deleted

**Review by Patton-L — CHANGES_REQUESTED** [review]

File: server/Pipfile (hunk 2 above)
Comment (mcgraths7) [reply server/Pipfile]: Added pipfile back to PR

File: server/.gitignore (hunk 1 above)
Comment (mcgraths7) [reply server/.gitignore]: removed from gitignore

File: server/Pipfile (hunk 2 above)
Comment (mcgraths7) [reply server/Pipfile]: I don't know the repercussions of rolling back to the commit where it still existed, will ask in standup tomorrow

**Review by Patton-L — APPROVED** [review]

**Review by cpascale43 — APPROVED** [review]

**Pull request merged**